wild_lib/src/layout.rs

```diff
@@ -2980,14 +2979,19 @@ impl<'data> ObjectLayoutState<'data> {
                 SectionSlot::Unloaded(unloaded) | SectionSlot::MustLoad(unloaded) => {
                     self.load_section(common, queue, *unloaded, section_id, resources)?;
                 }
+                SectionSlot::UnloadedDebugInfo(temp_part_id) => {
+                    self.load_debug_section(common, *temp_part_id, section_id, resources)?;
+                }
                 SectionSlot::Discard => {
                     bail!(
                         "{self}: Don't know what segment to put `{}` in, but it's referenced",
                         self.object.section_display_name(section_id),
                     );
                 }
-                SectionSlot::Loaded(_) | SectionSlot::EhFrameData(..) => {}
-                SectionSlot::MergeStrings(_) => {
+                SectionSlot::Loaded(_)
+                | SectionSlot::EhFrameData(..)
+                | SectionSlot::MergeStrings(_)
+                | SectionSlot::LoadedDebugInfo(..) => {
                     // We currently always load everything in merge-string sections. i.e. we don't
```

Comment:
Before, this comment was only in the `MergeStrings` arm of the match, but now it's in the arm for multiple rules. Perhaps move it before `| SectionSlot::MergeStrings(_)`?